src/app/file-manager/file-download-modal/file-download-modal.component.ts

```diff
@@ -92,4 +210,74 @@ export class FileDownloadModalComponent  {
       summary: warnMessage,
     });
   }
+
+  /**
+   * This function filters repeated folders when a parent
+   * is all-selected and there are sub-folders in the list,
+   * Items with indeterminate === true must be removed because
+   * selected content is included as other items.
+   * All this extra, non relevante here, items are required
+   * in the selection process.
+   */
+  filterFolderItems(sortedItems): Item[] {
```

Comment:
Declare types where possible, e.g. here this should be `sortedItems: Item[]`

Reply:
done.